acquia_lift.module

```diff
@@ -0,0 +1,262 @@
+ */
+function acquia_lift_ctools_render_alter(&$info, &$page, &$context) {
+  $task = $context['task'];
+  if ($page && $task['module'] == 'page_manager' && $task['name'] == 'node_view' && !empty($context['args'])) {
+    $nid = $context['args'][0];
+    if (isset($context['contexts']['argument_entity_id:node_1']) && $context['contexts']['argument_entity_id:node_1']->argument == $nid) {
+      $node = $context['contexts']['argument_entity_id:node_1']->data;
+      acquia_lift_get_page_context($node, TRUE);
+    }
+  }
+}
+
+/**
+ * Gets the page context data.
+ *
+ * @return
+ *   An array of page context data.
+ */
+function acquia_lift_get_page_context($node = NULL, $primary = FALSE) {
+  // Add any page context
+  module_load_include('inc', 'acquia_lift', 'acquia_lift.context');
+  $page_context = _acquia_lift_get_page_context($node, $primary);
+
+  if (!isset($page_context['content_title']) && is_null($node)) {
+    $page_context['content_title'] = drupal_get_title();
+  }
+
+  // Allow other modules to alter the page context to include additional data.
+  drupal_alter('acquia_lift_page_context', $page_context);
+  return $page_context;
+}
+
+/**
+ * Helper function to return a list of udf types.
+ *
+ * @return array
+ *   An array of UDF types defined in Acquia Lift Profile Manager with keys
+ *   as the type name and values as the number allowed.
+ */
+function acquia_lift_get_udf_types() {
+  return array(
+    'person' => 50,
+    'touch' => 20,
+    'event' => 20,
+  );
+}
+
+/**
+ * Returns the list of UDFs that can be mapped to.
+ *
+ * @param string $type
+ *   The type of udf to generate, one of: touch, person, event
+ *
+ * @return array
+ *   An array of UDFs defined in Acquia Lift Profile Manager.
+ */
+function acquia_lift_get_udfs($type) {
+  $counts = acquia_lift_get_udf_types();
+  if (!array_key_exists($type, $counts)) {
+    return array();
+  }
+  $count = $counts[$type];
+
+  for ($i = 1; $i <= $count; $i++) {
+    $udfs[] = $type . '_udf' . $i;
+  }
+  return $udfs;
+}
+
+function acquia_lift_get_grouped_context_options() {
+  module_load_include('inc', 'acquia_lift', 'acquia_lift.context');
+  $context_options = _acquia_lift_get_context_options();
+  foreach ($context_options as $context_type => $options) {
+    foreach ($options as $code => $info) {
+      $option_name = $context_type . ACQUIA_LIFT_ADMIN_SEPARATOR . $code;
+      if (isset($info['group'])) {
+        $group = $info['group'];
+        if (!isset($groups[$group])) {
+          $groups[$group] = array();
+        }
+        $groups[$group][$option_name] = $info['name'];
+      }
+      else {
+        $groups['Miscellaneous'][$option_name] = $info['name'];
+      }
+    }
+  }
+  return $groups;
+}
+
+function _acquia_lift_add_meta_tag($name, $value) {
+  $tag = array(
+    '#type' => 'html_tag',
+    '#tag' => 'meta',
+    '#attributes' => array(
+      'content' =>  $value,
+      'itemprop' => 'acquia_lift:' . $name,
```

Comment:
I wrote my tests to work in either order but have to admit I was puzzled why they were written out of order.  Thanks for pointing it out.  Darn drupal.